src/lib/SapphireClient.ts

```diff
@@ -19,10 +19,109 @@ import type { Awaited } from './utils/Types';
 // Extensions
 import './extensions/SapphireMessage';
 
+/**
+ * A valid prefix in Sapphire.
+ * * `string`: a single prefix, e.g. `'!'`.
+ * * `string[]`: an array of prefixes, e.g. `['!', '.']`.
+ * * `null`: disabled prefix, locks the bot's command usage to mentions only.
+ */
+export type SapphirePrefix = string | readonly string[] | null;
+
 export interface SapphirePrefixHook {
-	(message: Message): Awaited<string | readonly string[] | null>;
+	(message: Message): Awaited<SapphirePrefix>;
+}
+
+export interface SapphireClientOptions {
+	/**
+	 * The base user directory, if set to `null`, Sapphire will not call [[SapphireClient.registerUserDirectories]],
+	 * meaning that you will need to manually set each folder for each store. Please read the aforementioned method's
+	 * documentation for more information.
+	 * @since 1.0.0
+	 * @default undefined
+	 */
+	baseUserDirectory?: string | null;
+
+	/**
+	 * The default prefix, if it returns `null`, only mention prefix will trigger the bot's commands.
+	 * @since 1.0.0
+	 * @default null
+	 */
+	defaultPrefix?: SapphirePrefix;
+
+	/**
+	 * The prefix hook, by default it is a callback function that returns [[SapphireClientOptions.defaultPrefix]].
+	 * @since 1.0.0
+	 * @default () => client.options.defaultPrefix
+	 */
+	fetchPrefix?: SapphirePrefixHook;
+
+	/**
+	 * The internationalization options, defaults to an instance of [[Internationalization]] when
+	 * [[ClientInternationalizationOptions.instance]] is not specified.
+	 * @since 1.0.0
+	 * @default { instance: new Internationalization('en-US') }
+	 */
+	i18n?: ClientInternationalizationOptions;
+
+	/**
+	 * The client's ID, this is automatically set by the CoreReady event.
+	 * @since 1.0.0
+	 * @default this.client.user?.id ?? null
+	 */
+	id?: string;
+
+	/**
+	 * The logger options, defaults to an instance of [[Logger]] when [[ClientLoggerOptions.instance]] is not specified.
+	 * @since 1.0.0
+	 * @default { instance: new Logger(LogLevel.Info) }
+	 */
+	logger?: ClientLoggerOptions;
 }
 
+/**
+ * The base [[Client]] extension that makes Sapphire work. When building a Discord bot with the framework, the developer
+ * must either use this class, or extend it.
+ *
+ * Sapphire also automatically detects the folders to scan for pieces, please read
+ * [[SapphireClient.registerUserDirectories]] for reference. This method is called at the start of the
+ * [[SapphireClient.login]] method, however, you can change the root or opt-out.
```

Comment:
```suggestion
 * [[SapphireClient.login]] method.
```

This is out-of-place here as it is documented in the method's description. Also the sentence was just plain wrong grammatically with it.